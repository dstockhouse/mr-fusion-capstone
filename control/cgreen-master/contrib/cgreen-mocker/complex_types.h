typedef struct BasicStruct {
    int someValue;
} BasicStruct;

typedef BasicStruct* BasicStructPtr;

BasicStruct return_struct_by_value(int i);
BasicStructPtr return_pointer_to_struct(char string[]);
BasicStruct* direct_return_pointer_to_struct(void);
