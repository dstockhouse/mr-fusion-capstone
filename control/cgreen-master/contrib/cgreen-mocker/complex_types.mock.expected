/* -*- c -*-*/
#include "complex_types.h"
#include <cgreen/mocks.h>

BasicStruct return_struct_by_value(int i) { 
  return *(BasicStruct *) mock(i);
}

BasicStructPtr return_pointer_to_struct(char string[]) { 
  return (BasicStructPtr) mock(string);
}

BasicStruct *direct_return_pointer_to_struct(void) { 
  return (BasicStruct *) mock();
}

